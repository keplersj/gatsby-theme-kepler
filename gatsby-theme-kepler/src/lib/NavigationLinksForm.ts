interface Item {
  name: string;
}

export const NavigationLinksForm = {
  label: "Navigation Links",
  fields: [
    {
      label: "Navigation Links",
      name: "rawJson.navLinks",
      component: "group-list",
      itemProps: (item: Item) => ({
        key: item.name,
        label: item.name,
      }),
      fields: [
        {
          label: "Name",
          name: "name",
          component: "text",
        },
        {
          label: "Link",
          name: "url",
          component: "text",
        },
      ],
    },
  ],
};
