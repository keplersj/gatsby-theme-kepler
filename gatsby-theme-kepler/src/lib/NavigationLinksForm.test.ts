import { NavigationLinksForm } from "./NavigationLinksForm";

describe("NavigationLinksForm", () => {
  it("returns as expected", () => {
    expect(NavigationLinksForm).toMatchSnapshot();
  });

  describe("field props", () => {
    it("returns the props of field item correctly", () => {
      const field = NavigationLinksForm.fields[0];

      const testItem = {
        name: "Test Name",
        url: "https://example.dev/",
      };

      expect(field.itemProps(testItem)).toMatchInlineSnapshot(`
        Object {
          "key": "Test Name",
          "label": "Test Name",
        }
      `);
    });
  });
});
