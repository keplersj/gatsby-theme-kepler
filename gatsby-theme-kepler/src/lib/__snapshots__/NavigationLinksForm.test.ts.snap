// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationLinksForm returns as expected 1`] = `
Object {
  "fields": Array [
    Object {
      "component": "group-list",
      "fields": Array [
        Object {
          "component": "text",
          "label": "Name",
          "name": "name",
        },
        Object {
          "component": "text",
          "label": "Link",
          "name": "url",
        },
      ],
      "itemProps": [Function],
      "label": "Navigation Links",
      "name": "rawJson.navLinks",
    },
  ],
  "label": "Navigation Links",
}
`;
